### ISSUE TITLE
How to use LineGraph

### ISSUE THREAD

**Issue by PPGGG** [pyg-team/pytorch_geometric#5349]
### 🐛 Describe the bug

I would like to change line classification problem as node classification problem, And it seems that `LineGraph` could solve the problem.

But i would like to know is whether the line classification problem in `LineGraph` is the same as original, which aims to judge whether a line between the given two nodes ?

**Comment by a-Alarcon99** [comment]
Hi PPGGG, I have been doing some research about the project and if we notice, in the LineGraph class there is a reference to the torhgeometric/data.py package, in which all the data is managed (including the variable 'y') . You can try to implement it with a very basic example to see if it works.
You can take a look at the class from here: [Data.py](https://github.com/pyg-team/pytorch_geometric/blob/c8e1cdb4665d408dd2d5de3fbe0fb2fdb0bfbd1d/torch_geometric/data/data.py#L303)

**Comment by PPGGG** [comment]
@a-Alarcon99 Thanks. I have tried to convert original graph to LineGraph, like below:

```
path = "./data/Amazon"
dataset = Amazon(path, name='Photo')
transform = RandomNodeSplit(split='train_rest',
                            num_val=0.2, num_test=0.2)
photo_data = transform(dataset[0])
print(photo_data)

[Output] Data(x=[7650, 745], edge_index=[2, 238162], y=[7650], train_mask=[7650], val_mask=[7650], test_mask=[7650])
```
Then i convert `photo_data` to `linegraph_data` using `LineGraph`.

```
linegraph_data = LineGraph()(photo_data)
print(linegraph_data)

[Output] Data(x=[7650, 745], edge_index=[2, 24273186], y=[7650], train_mask=[7650], val_mask=[7650], test_mask=[7650], num_nodes=119081)
```

Through the operation, it seems that the `num_nodes` in `x` is not equal to the `num_nodes` which should be `119081` in the `linegraph_data`. 

If you could give me some hint, it would be really appreciated.

**Comment by a-Alarcon99** [comment]
Okay, I will try to provide you a solution. I have been looking at your example code and it seems that the number of nodes that are created in the `LineGraph` is defined in `edge_index` of `photo_data`. The number of total nodes will be `edge_index[1]`/`edge_index[0]`, so in this case, and as we can see in `linegraph_data`, the number of nodes is 119081:


```python
print(photo_data)

[Output] Data(x=[7650, 745], edge_index=[2, 238162], y=[7650], train_mask=[7650], val_mask=[7650], test_mask=[7650])
```

`edge_index=[2, 238162]`
With this example you have to divide 238162 by 2 to obtain 119081.

**Comment by PPGGG** [comment]
@a-Alarcon99 Thanks a lot. Now i wonder how to get node attribute in `LineGraph`. It would be great if you could provide more example. Thanks a lot.

**Comment by a-Alarcon99** [comment]
I am not 100 percent sure how it works but I am not seeing any attribute called Node in `pytorch_geometric/line_graph.py`. 
The attributes of the LineGraph class can be seen in the output:
```
linegraph_data = LineGraph()(photo_data)
print(linegraph_data)

[Output] Data(x=[7650, 745], edge_index=[2, 24273186], y=[7650], train_mask=[7650], val_mask=[7650], test_mask=[7650], num_nodes=119081)
```

The attributes are: `x`, `edge_index`, `y`, `train_mask`, `val_mask`, `test_mask`, `num_nodes`

**Comment by PPGGG** [comment]
@a-Alarcon99 That's true. But it seems that we could not directly finish the link prediction task through `LineGraph`.

**Comment by a-Alarcon99** [comment]
@PPGGG So have you tried to implement `linegraph_data` with your classification model and it hasn't worked?

**Comment by PPGGG** [comment]
@a-Alarcon99  Yes i have tried. But i stuck with the situation that i don't know how to prepare the node feature and label information in `linegraph`, which are essential to classification.

**Comment by PPGGG** [comment]
@a-Alarcon99 I have found the code that may close to solve my question. But the preprocess is too complex.

**Comment by a-Alarcon99** [comment]
@PPGGG Good to hear you found a solution! Can you share it so that I can observe the process that is carried out to obtain the correct implementation?

**Comment by PPGGG** [comment]
@a-Alarcon99 I am still working on that. I found it hard to really understand the code.
